main.tf

```diff
@@ -0,0 +1,141 @@
+resource "aws_s3_bucket" "data" {
```

Comment:
<details><summary>
<img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsSeverityIcons/low_new.svg" alt="LOW" width="24px" align="center">&nbsp;&nbsp;<b style="color:black" >Ensure S3 buckets are encrypted with KMS by default</b>
            <br>
            &nbsp;&nbsp;&nbsp;&nbsp;Resource: <a href="https://github.com/marynaKK/terragoat/blob/marynaKK-patch-1/main.tf#L1-L21">aws_s3_bucket.data</a> | Bridgecrew ID: <code>BC_AWS_GENERAL_56</code> | Checkov ID: <a href="https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/aws-general-policies/ensure-that-s3-buckets-are-encrypted-with-kms-by-default">CKV_AWS_145</a>
            <br>
            
</summary>
<h4>How to Fix</h4>

```tf
resource "aws_s3_bucket" "bucket_name" {
  bucket = "bucket_good"
}

+ resource "aws_s3_bucket_server_side_encryption_configuration" "good_sse_1" {
+   bucket = aws_s3_bucket.bucket_name.bucket
+
+   rule {
+     apply_server_side_encryption_by_default {
+       kms_master_key_id = aws_kms_key.mykey.arn
+       sse_algorithm     = "aws:kms"
+     }
+   }
+ }
```

<h4>Description</h4>


Encrypting your data and resources with KMS helps protect your data from unauthorized access or tampering.
By encrypting your data, you can ensure that only authorized users can access and decrypt the data, and that the data is protected while in storage or in transit.
Such action can help protect against external threats such as hackers or malware, as well as internal threats such as accidental or unauthorized access.

</details>
<p><img src="https://bc-network-graph-icons.s3.us-west-2.amazonaws.com/prCommentsBotLogos/bc.svg" alt="Bridgecrew" width="18px">&nbsp;<small>Comment by Bridgecrew Bot</small></p>